TestProjects/ShaderGraph/Assets/CommonAssets/Editor/ShaderStageCapabilityTests.cs

```diff
@@ -0,0 +1,168 @@
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+using UnityEditor.Graphing;
+using UnityEditor.Graphing.Util;
+using UnityEditor.ShaderGraph.Serialization;
+
+
+namespace UnityEditor.ShaderGraph.UnitTests
+{
+    [TestFixture]
+    internal class ShaderStageCapabilityTests
+    {
+        static string targetUnityDirectoryPath => "Assets/Testing/ShaderStageCapabilityGraphs";
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+        }
+
+        [OneTimeTearDown]
+        public void Cleanup()
+        {
+        }
+
+        NodeType FindFirstNodeOfType<NodeType>(GraphData graphData, string nodeName = null) where NodeType : AbstractMaterialNode
+        {
+            var nodes = graphData.GetNodes<NodeType>();
+            foreach (var node in nodes)
+            {
+                if (nodeName == null || node.name == nodeName)
+                    return node;
+            }
+            return null;
+        }
+
+        MaterialSlot FindNamedSlot(AbstractMaterialNode node, string slotName)
+        {
+            foreach (var slot in node.GetSlots<MaterialSlot>())
+            {
+                if (slot.RawDisplayName() == slotName)
+                    return slot;
+            }
+            return null;
+        }
+
+        [Test]
+        public void SubGraphDescendentsTests()
+        {
+            var graphPath = targetUnityDirectoryPath + "/ShaderStageCapability_Graph.shadergraph";
+
+            string fileContents = File.ReadAllText(graphPath);
+            var graphGuid = AssetDatabase.AssetPathToGUID(graphPath);
+            var messageManager = new MessageManager();
+            GraphData graphData = new GraphData() { assetGuid = graphGuid, messageManager = messageManager };
+            MultiJson.Deserialize(graphData, fileContents);
+            graphData.OnEnable();
+            graphData.ValidateGraph();            
+
+            var subGraphnodeName = "ShaderStageCapability_SubGraph";
+            var subGraphNode = FindFirstNodeOfType<SubGraphNode>(graphData, subGraphnodeName);
+            if(subGraphNode == null)
```

Comment:
I would have thought this would trip formatting (maybe formatting skips the TestProjects), but if the job didn't flag it, I wouldn't bother changing it.

Reply:
Formatting does seem to skip TestProjects. I had to update to latest master anyways so I fixed this while I was there.